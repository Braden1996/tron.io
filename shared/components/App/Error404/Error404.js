import React, { Component, PropTypes } from 'react';

class Error404 extends Component {
  componentWillMount() {
    const { staticContext } = this.context.router;
    if (staticContext) {
      staticContext.missed = true;
    }
  }

  render() {
    return (
      <div>Sorry, that page was not found.</div>
    );
  }
}

Error404.contextTypes = {
  router: PropTypes.shape({
    staticContext: PropTypes.object,
  }).isRequired,
};

export default Error404;
