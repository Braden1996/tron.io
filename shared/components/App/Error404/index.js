import React, { Component, PropTypes } from 'react';

class Error404 extends Component {
  componentWillMount() {
    const { staticContext } = this.props;
    if (staticContext) {
      staticContext.missed = true;
    }
  }

  render() {
    return (
      <div>Sorry, that page was not found.</div>
    );
  }
}

Error404.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  staticContext: PropTypes.object.isRequired,
};

export default Error404;
