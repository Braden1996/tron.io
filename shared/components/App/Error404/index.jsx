import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Error404 extends Component {
  componentWillMount() {
    const { staticContext } = this.props;
    if (staticContext) {
      staticContext.missed = true;
    }
  }

  render() {
    return <div>Sorry, that page was not found.</div>;
  }
}

Error404.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  staticContext: PropTypes.object,
};

Error404.defaultProps = {
  staticContext: {},
};

export default Error404;
