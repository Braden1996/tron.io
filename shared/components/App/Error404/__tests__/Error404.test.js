/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { shallow } from 'enzyme';

import Error404 from '../index';

describe('<Error404 />', () => {
  test('renders', () => {
    const staticContext = {};
    const wrapper = shallow(<Error404 staticContext={staticContext} />);
    expect(wrapper).toMatchSnapshot();
    expect(staticContext.missed).toBeTruthy();
  });
});
