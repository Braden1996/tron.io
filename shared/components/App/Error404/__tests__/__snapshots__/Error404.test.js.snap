// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error404 /> renders 1`] = `
<div>
  Sorry, that page was not found.
</div>
`;
